js/index.js

```diff
@@ -42,42 +57,119 @@ function renderPepperoni() {
   });
 }
 
+
+
+// Iteration 1: set the visibility of `<section class="mushroom">`
 function renderMushrooms() {
-  // Iteration 1: set the visibility of `<section class="mushroom">`
+  document.querySelectorAll('.mushroom').forEach(function (oneMushroom) {
+    if (state.mushrooms === true) {
+      oneMushroom.style.visibility = 'visible';
+    } else {
+      oneMushroom.style.visibility = 'hidden';
+    }
+  });
+
 }
 
 function renderGreenPeppers() {
   // Iteration 1: set the visibility of `<section class="green-pepper">`
+  document.querySelectorAll('.green-pepper').forEach(function (onePepper) {
+    if (state.greenPeppers) {
+      onePepper.style.visibility = 'visible';
+    } else {
+      onePepper.style.visibility = 'hidden';
+    }
+  })
+
+
+
 }
 
+
+
+
+
 function renderWhiteSauce() {
   // Iteration 2: add/remove the class "sauce-white" of `<section class="sauce">`
+  document.querySelectorAll('.sauce').forEach(function (sauce) {
+    if (state.whiteSauce === false) {
+      sauce.removeAttribute("sauce-white");
+    } else {
+      sauce.setAttribute("class", "sauce-white");
+    }
+  })
+
+
 }
 
 function renderGlutenFreeCrust() {
   // Iteration 2: add/remove the class "crust-gluten-free" of `<section class="crust">`
+
+  document.querySelectorAll('.crust').forEach(function (crust) {
+    if (state.glutenFreeCrust === false) {
+      crust.removeAttribute("crust-gluten-free");
+    } else {
+      sauce.setAttribute("class", "crust-gluten-free");
+    }
+  })
+
+
+
+
 }
```

Comment:
On this both functions you could have used `querySelector` because there is just one element `.sauce` and one `.crust`, another thing is that you should have used the method `classList.add` and `classList.remove`

`function renderWhiteSauce() { const oneSauce = document.querySelector(".sauce"); if (state.whiteSauce) { oneSauce.classList.add("sauce-white"); } else { oneSauce.classList.remove("sauce-white"); } }`

`function renderGlutenFreeCrust() { const oneCrust = document.querySelector(".crust"); if (state.glutenFreeCrust) { oneCrust.classList.add("crust-gluten-free"); } else { oneCrust.classList.remove("crust-gluten-free"); } }`